src/compiler/program.ts

```diff
@@ -1734,6 +1734,7 @@ namespace ts {
                 if (!newSourceFile) {
                     return StructureIsReused.Not;
                 }
+                newSourceFile.impliedNodeFormat = sourceFileOptions.impliedNodeFormat;

```

Comment:
Isn't the `getSourceFile` call on the prior lines supposed to handle this? It uses the options impliedNodeFormat to set weather a new source file is supposed to be returned by the call.